docs/guides/git-actions.md

```diff
@@ -0,0 +1,140 @@
+## Setting up CI/CD with [Nitric](https://nitric.io) and [GitHub actions](https://github.com/features/actions)
+
+GitHub has comprehensive documentation which can guide you through advanced workflows.
+
+Here we'll walk through the simple example of deploying a Nitric project to AWS.
+
+## Lets start by getting our workflow setup
+
+In your project you'll want to create a yaml file which contains your workflow config. You can name this file anything you like, in this example we've gone with 'deploy-aws.yaml'.
+
+It should be placed the folder `.github/workflows`
+
+## Full example
+
+```yaml
+name: Sample configuration to deploy to AWS
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+env:
+  PULUMI_CONFIG_PASSPHRASE: ${{ secrets.PULUMI_ACCESS_TOKEN }}
+  PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
+jobs:
+  update:
+    name: Update
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - name: Install Nitric CLI
+        uses: nitrictech/setup-nitric@v1
+        with:
+          version: 1.2.1
+      - name: Install dependencies
+        uses: pulumi/setup-pulumi@v2
+      - name: Checkout project
+        uses: actions/checkout@v3
+      - name: Resolve packages
+        run: npm install
+      - name: Deploy stack to aws
+        run: nitric up -s dev -v0
+```
+
+## Breaking it down
+
+Edit the config file and start by defining a name.
+
+```yaml
+name: Sample configuration to deploy to AWS
+```
+
+### Setup action triggers
+
+Action triggers tell your workflow when to run.
+
+- workflow_dispatch
+  - This trigger allows the workflow to be manually run from GitHub
+- push -> branches -> main
+  - This will trigger this workflow each time a push is performed on the main branch
+
+```yaml
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+```
+
```

Comment:
Since setup action triggers got changed to a h3 heading, should there be a few more h3 headings to separate the content? i.e. one here to say ### Configuring environment variables

Reply:
Updated